### PR TITLE
add terminal controller and lane controller scripts

### PR DESCRIPTION
**Pull request by ashanhol**
Solves #1 and #13

### CODE REVIEW

File: Assets/Scripts/LaneController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        isHacked = false;
+        isHacking = false;
+        hackTimeElapsed = 0;
+    }
+
+
+    // Update is called once per frame
+    void Update () {
+        
+        if (!isHacked)
```
Comment (andrew-ws) [inline Assets/Scripts/LaneController.cs]: Add brackets here

File: Assets/Scripts/LaneController.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            //If opponent minion collides with terminal
+            if (isHacking)
+            {
+                Transmitting();
+            }
+
+    }
+
+    //process of hacking. call by terminal after minion collides
+    public void Transmitting()
```
Comment (andrew-ws) [inline Assets/Scripts/LaneController.cs]: Call this UpdateHacking or something like that

File: Assets/Scripts/LaneController.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            isHacking = true;
+
+        hackTimeElapsed += Time.deltaTime;
+
+        if (hackTimeElapsed >= HackTime)
+            isHacked = true;
+    }
+
+    //call this to stop hacking if minion dies 
+    public void SeverConnection()
```
Comment (andrew-ws) [inline Assets/Scripts/LaneController.cs:52]: can this be on the terminal

File: Assets/Scripts/TerminalController.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using GG18.Minions.Minion;
+
+public class TerminalController : MonoBehaviour {
+
+    public Transform MissileSpawn;
```
Comment (andrew-ws) [inline Assets/Scripts/TerminalController.cs:8]: should this be a vector3?

File: Assets/Scripts/TerminalController.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+            if (!laneController.isHacked)
+            {
+                laneController.HackedBy = PlayerIndex;
+                laneController.Transmitting();
+            }
+        }
+    }
+
+    //If the minion from opposite team dies, sever connection.
+    private void OnTriggerExit(Collider other)
```
Comment (andrew-ws) [inline Assets/Scripts/TerminalController.cs:36]: Will this trigger on destroy? I would think it would only trigger when something moves out

Comment (andrew-ws) [reply Assets/Scripts/TerminalController.cs:36]: It probably does but maybe factor this into a helper method just in case

**Review by andrew-ws — DISMISSED** [review]

File: Assets/Scripts/LaneController.cs (hunk 1 above)
Comment (ashanhol) [reply Assets/Scripts/LaneController.cs]: Done

File: Assets/Scripts/LaneController.cs (hunk 2 above)
Comment (ashanhol) [reply Assets/Scripts/LaneController.cs]: Done

File: Assets/Scripts/LaneController.cs (hunk 3 above)
Comment (ashanhol) [reply Assets/Scripts/LaneController.cs:52]: Done

File: Assets/Scripts/TerminalController.cs (hunk 4 above)
Comment (ashanhol) [reply Assets/Scripts/TerminalController.cs:8]: Done

File: Assets/Scripts/TerminalController.cs (hunk 5 above)
Comment (ashanhol) [reply Assets/Scripts/TerminalController.cs:36]: Done

**Pull request merged**